superset-frontend/src/visualizations/TimeTable/FormattedNumber.tsx

```diff
@@ -17,27 +17,23 @@
  * under the License.
  */
 import React from 'react';
-import PropTypes from 'prop-types';
 import { formatNumber } from '@superset-ui/core';
 
-const propTypes = {
-  num: PropTypes.number,
-  format: PropTypes.string,
-};
-
-const defaultProps = {
-  num: 0,
-  format: undefined,
-};
+interface FormattedNumberProps {
+  num?: string | number;
+  format?: string;
+}
 
-function FormattedNumber({ num, format }) {
+function FormattedNumber({
+  num = 0,
+  format = undefined,
+}: FormattedNumberProps) {
   if (format) {
-    return <span title={num}>{formatNumber(format, num)}</span>;
+    return (
+      <span title={num as string}>{formatNumber(format, num as number)}</span>
```

Comment:
it looks like `formatNumber` might actually support strings already (according to the old code). could you check the function in superset-ui and see if that type needs to be changed?

also, for `title`, let's do ```title={`${num}`}``` instead asserting and overriding the type

Reply:
> it looks like `formatNumber` might actually support strings already (according to the old code). could you check the function in superset-ui and see if that type needs to be changed?

Sure thing. 

> also, for `title`, let's do `` title={`${num}`} `` instead asserting and overriding the type

Will do.

Reply:
@etr2460 

> it looks like `formatNumber` might actually support strings already (according to the old code). could you check the function in superset-ui and see if that type needs to be changed?
> 
As per the[ superset-ui documentation for number-format](https://github.com/apache-superset/superset-ui/tree/master/packages/superset-ui-core/src/number-format), it looks like `formatNumber`  fallbacks to `d3.format` to handle the error for invalid format like `string`.

However, changing: 

`<span title={num as string}>{formatNumber(format, num as number)}</span>`

to 

`<span title={num as string}>{formatNumber(format, num)}</span>;`

in the tsx file is throwing an error: 

_Argument of type 'string | number' is not assignable to parameter of type 'number | null | undefined'.
  Type 'string' is not assignable to type 'number | null | undefined'._

Reply:
@Damans227 @etr2460 Just to be sure here. The [superset-ui documentation for number-format](https://github.com/apache-superset/superset-ui/tree/master/packages/superset-ui-core/src/number-format) referenced above states the following:

> It is powered by a registry to support registration of custom formatting, with fallback to d3.format and handle error for invalid format string.

It's talking about the formatting string, which is the first parameter of the function, and not about invalid formats for the `num` (second parameter). Are you sure the value can be interpreted as strings?

Reply:
@michael-s-molina @Damans227 so this uses d3-format under the hood, which is untyped, but as far as i can tell does support strings (see where it converts it here):
https://github.com/d3/d3-format/blob/main/src/locale.js#L74

But more importantly than that, before we allowed all strings through here, so i don't think adding a `parseInt` is correct since that could change the behavior. coercing the type with an `as number` also doesn't feel right since we don't know it's a number. That's why i recommended the options above